### ISSUE TITLE
Developers documentation

### ISSUE THREAD

**Issue by pavlenex** [btcpayserver/btcpayserver-doc#174]
At some point we should add developers documentation section in docs to encourage people to contribute and make it easier for them to get around.

I know @britttttk has some excellent docs which are WIP and we can add the [developing on BTCPay ](https://www.youtube.com/watch?v=ZePbMPSIvHM)Video and also Rockstar's [talk](https://www.youtube.com/watch?v=jwJV5gsmtFM) somewhere.

**Comment by britttttk** [comment]
Great idea.

Nicolas has very thoughtfully made it easy to develop and test BTCPay with docker-compose! 
He already wrote [how to be started for development](https://github.com/btcpayserver/btcpayserver/blob/master/BTCPayServer.Tests/README.md) and created the developer video. 

Hopefully Rockstardev will record his upcoming BTCPay dev meetup on Feb. 6 as well. 

So IMO the technical information already exists. However, I think it could be helpful to create a really general document to link it all together. Also a link to the [GitHub Hello World Tutorial](https://guides.github.com/activities/hello-world/) might be helpful for encouraging new developers. Or how to [test someone else's pull request](https://help.github.com/articles/checking-out-pull-requests-locally/). 

I'm not really sure who the audience is for this but it would be nice to have some documentation in the docs repo related to development.

**Comment by britttttk** [comment]
We could possibly add the information to [LocalDevelopment.md](https://github.com/btcpayserver/btcpayserver-doc/blob/master/LocalDevelopment.md)

**Comment by Eskyee** [comment]
Great idea...

**Comment by pavlenex** [comment]
I don't think #316 closes this issue, there's a ton of work we can do to actually make things easier for developers. I'll re-open this one. First and foremost, it will be useful once the new API is done.

**Comment by dennisreimann** [comment]
Just as a reference and update: This relates to #501 and #545.

**Comment by rockstardev** [comment]
Sorry all, I'll explain why I removed my assignment next time we drink together.
